crates/compiler/gen_dev/src/lib.rs

```diff
@@ -1085,12 +1085,12 @@ trait Backend<'a> {
                     args.len(),
                     "NotEq: expected to have exactly two argument"
                 );
-                debug_assert_eq!(
-                    arg_layouts[0], arg_layouts[1],
+                debug_assert!(
+                    self.interner().eq_repr(arg_layouts[0], arg_layouts[1]),
                     "NotEq: expected all arguments of to have the same layout"
                 );
                 debug_assert_eq!(
-                    Layout::BOOL,
+                    Layout::BOOL_SEMA,
                     *ret_layout,
                     "NotEq: expected to have return layout of type Bool"
                 );
```

Comment:
we could do `self.interner().eq_repr(arg_layouts[0], Layout::BOOL)` here instead, which would pass.